types/index.d.ts

```diff
@@ -0,0 +1,160 @@
+import type { Client, Subscription } from 'aedes';
+import type { AedesPacket } from 'aedes-packet';
+import type { QoS } from 'mqtt-packet';
+import type { Readable } from 'stream';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId: string;
+  topic: string,
+  qos?: QoS
+}
```

Comment:
why this? Shouldn't this be in aedes itself?

Reply:
Same as above, currently there is the `Subscription` interface which comes from `mqtt-packet` but it doesn't contain the clientId property. 
Also my intent was to have an internal interface that describes the content of internally stored subscription.
Once Aedes is updated, i will probably create an interface that just picks those 3 properties (clientId, topic and qos), from aedes.Subscription.

Reply:
I think it's better to send the relevant PR to aedes first and maybe to mqtt-packet if the field is actually there.

Reply:
In fact that was a mistake, the subscription does not contain a `clientId`. I got misled by `outgoingEnqueue` argument called `sub` which is expected to be an object containing the `clientId`.